gitrevise/odb.py

```diff
@@ -274,10 +306,43 @@ def new_commit(
             body += b"parent " + parent.oid.hex().encode() + b"\n"
         body += b"author " + author + b"\n"
         body += b"committer " + committer + b"\n"
-        body += b"\n"
-        body += message
+
+        body_tail = b"\n" + message
+        body += self.sign_buffer(body + body_tail)
+        body += body_tail
+
         return Commit(self, body)
 
+    def sign_buffer(self, buffer: bytes) -> bytes:
+        """Return the text of the signed commit object."""
+        if not self.sign_commits:
+            return b""
+
+        key_id = self.config(
+            "user.signingKey", default=self.default_committer.signing_key
+        )
+        gpg = None
+        try:
+            gpg = run(
+                (self.gpg, "--status-fd=2", "-bsau", key_id),
+                stdout=PIPE,
+                stderr=PIPE,
+                input=buffer,
+                check=True,
+            )
+        except CalledProcessError as gpg:
+            print(gpg.stderr.decode(), file=sys.stderr, end="")
```

Comment:
This access to `gpg.stderr` will never work: if `run()` throws, `gpg` will not have been assigned a value other than the `None` above.

Reply:
Nevermind, I see now that `gpg` is aliased to the caught exception itself!

Reply:
heh, that's pretty clever :open_mouth: